src/software/simulated_tests/validation/non_terminating_function_validator_test.cpp

```diff
@@ -24,94 +25,40 @@ TEST(NonTerminatingFunctionValidatorTest,
     }
 }
 
-TEST(NonTerminatingFunctionValidatorTest,
-     test_world_updated_correctly_between_validation_function_restarts)
+TEST(NonTerminatingFunctionValidatorTest, test_yielding_error_message)
 {
     // This validation_functions uses exceptions as a way for the test to observe it's
-    // internal state
+    // internal state The exception will not be reached until the 3rd function call
     ValidationFunction validation_function = [](std::shared_ptr<World> world,
                                                 ValidationCoroutine::push_type& yield) {
-        if (world->ball().position().x() < -1)
-        {
-            throw std::runtime_error("x < -1");
-        }
-        else if (world->ball().position().x() < 0)
-        {
-            throw std::runtime_error("x < 0");
-        }
-        else if (world->ball().position().x() < 1)
-        {
-            throw std::runtime_error("x < 1");
-        }
-        return;
+        yield("This is an error message 1");
+        yield("This is an error message 2");
```

Comment:
What is the purpose of two error messages here?

Reply:
these are tests for the validator and we want to distinguish between the first and second error message